mempool/src/pool.rs

```diff
@@ -0,0 +1,1092 @@
+use std::cell::Cell;
```

Comment:
This file is 1000+ lines. Can we move some stuff to some submodules with good names to have smaller files?